pkg/diskmaker/controllers/lvset/reconcile.go

```diff
@@ -119,24 +166,39 @@ func (r *ReconcileLocalVolumeSet) Reconcile(request reconcile.Request) (reconcil
 		if lvset.Spec.MaxDeviceCount != nil {
 			withinMax = int32(alreadyProvisionedCount) < *lvset.Spec.MaxDeviceCount
 		}
-		if !withinMax {
+		// skip this device if this device is not already symlinked and provisioning it would exceed the maxDeviceCount
+		if !(withinMax || currentDeviceSymlinked) {
 			break
 		}
 
-		devLogger.Info("symlinking")
-		r.eventReporter.Report(lvset, newDiskEvent(diskmaker.FoundMatchingDisk, "symlinking matching disk", blockDevice.KName, corev1.EventTypeNormal))
-		err = symLinkDisk(lvset, r.eventReporter, devLogger, blockDevice, symLinkDir)
+		// Retrieve list of mount points to iterate through discovered paths (aka files) below
+		mountPoints, err := r.runtimeConfig.Mounter.List()
+		if err != nil {
+			return reconcile.Result{}, fmt.Errorf("error retrieving mountpoints: %w", err)
+		}
+		// Put mount points into set for faster checks below
+		type empty struct{}
```

Comment:
nothing uses this `empty` type afaict.

Reply:
yes, switched it to sets.String, will get rid of it.